indigo_app/views/tasks.py

```diff
@@ -388,8 +393,42 @@ def post(self, request, *args, **kwargs):
 
         task.save()
 
+        # if a conversion task has been finished, do the import
+        if task.code in ['convert-document'] and self.change in ['finish'] and task.work is not None:
+            self.create_and_import_document(task)
+
         return redirect(self.get_redirect_url())
 
+    def create_and_import_document(self, task):
```

Comment:
Would it make more sense to have this on the task object itself? Alternatively, all the logic for connecting tasks together could be on the task broker object, since it already has to do some of those things.